asv/plugins/mercurial.py

```diff
@@ -106,7 +106,7 @@ def checkout_existing():
             with hglib.open(path) as subrepo:
                 subrepo.pull()
                 subrepo.update(commit_hash, clean=True)
```

Comment:
Ah -- I see.  `clean=True` here only discards uncommitted changes, not untracked files.  This makes sense, then, and brings the behavior inline with the `git` plugin.